src/components/Sheet/Sheet.tsx

```diff
@@ -0,0 +1,180 @@
+import * as React from 'react';
+
+import {CSSTransition} from 'react-transition-group';
+import debounce from 'lodash/debounce';
+import {classNames} from '@shopify/react-utilities/styles';
+
+import {navigationBarCollapsed} from '../../utilities/breakpoints';
+import {Key} from '../../types';
+import {layer, overlay, Duration} from '../shared';
+import {frameContextTypes, FrameContext} from '../Frame';
+import {withAppProvider, WithAppProviderProps} from '../AppProvider';
+
+import Backdrop from '../Backdrop';
+import TrapFocus from '../TrapFocus';
+import Portal from '../Portal';
+import KeypressListener from '../KeypressListener';
+import EventListener from '../EventListener';
+
+import styles from './Sheet.scss';
+
+export const BOTTOM_CLASS_NAMES = {
+  enter: classNames(styles.Bottom, styles.enterBottom),
+  enterActive: classNames(styles.Bottom, styles.enterBottomActive),
+  exit: classNames(styles.Bottom, styles.exitBottom),
+  exitActive: classNames(styles.Bottom, styles.exitBottomActive),
+};
+
+export const RIGHT_CLASS_NAMES = {
+  enter: classNames(styles.Right, styles.enterRight),
+  enterActive: classNames(styles.Right, styles.enterRightActive),
+  exit: classNames(styles.Right, styles.exitRight),
+  exitActive: classNames(styles.Right, styles.exitRightActive),
+};
```

Comment:
I think we definitely could, but I feel like it would be at the expense of readability and searchability. It would involve string interpolation, which would make those classes inherently hard to search for, and hard to read. Not sure it's worth it for the sake of dryness